packages/shared/src/components/modals/OnboardingModal.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ReactElement, ReactNode, useContext, useState } from 'react';
+import { useQueryClient } from 'react-query';
+import { ModalProps } from './StyledModal';
+import SteppedModal from './SteppedModal';
+import useFeedSettings, {
+  getFeedSettingsQueryKey,
+  updateLocalFeedSettings,
+} from '../../hooks/useFeedSettings';
+import SettingsContext from '../../contexts/SettingsContext';
+import ThemeOnboarding from '../onboarding/ThemeOnboarding';
+import FilterOnboarding from '../onboarding/FilterOnboarding';
+import LayoutOnboarding from '../onboarding/LayoutOnboarding';
+import IntroductionOnboarding from '../onboarding/IntroductionOnboarding';
+import FeaturesContext from '../../contexts/FeaturesContext';
+import { OnboardingStep } from '../onboarding/common';
+import useTagAndSource from '../../hooks/useTagAndSource';
+import { useMyFeed } from '../../hooks/useMyFeed';
+import { AllTagCategoriesData } from '../../graphql/feedSettings';
+import AuthContext from '../../contexts/AuthContext';
+import { LoginTrigger } from '../../lib/analytics';
+
+const STEPS_COUNT = 4;
+
+interface OnboardingModalProps extends ModalProps {
+  trigger?: string;
+  onRegistrationSuccess?: () => void;
+}
+
+function OnboardingModal({
+  onRegistrationSuccess,
+  onRequestClose,
+  ...props
+}: OnboardingModalProps): ReactElement {
+  const client = useQueryClient();
+  const { refetchBoot } = useContext(AuthContext);
+  const { registerLocalFilters } = useMyFeed();
+  const [selectedTopics, setSelectedTopics] = useState({});
+  const [invalidMessage, setInvalidMessage] = useState<string>(null);
+  const { onboardingSteps, onboardingFiltersLayout, onboardingMinimumTopics } =
+    useContext(FeaturesContext);
+  const { insaneMode, themeMode, setTheme, toggleInsaneMode } =
+    useContext(SettingsContext);
+  const { tagsCategories } = useFeedSettings();
+  const { onFollowTags, onUnfollowTags } = useTagAndSource({
+    origin: 'filter onboarding',
+  });
+  const [step, setStep] = useState(0);
+  const onStepChange = (beforeStep: number, stepNow: number) => {
+    setStep(stepNow);
+    const filterStep = onboardingSteps.indexOf(OnboardingStep.Topics);
+
+    if (beforeStep - 1 === filterStep) {
+      const key = getFeedSettingsQueryKey();
+      const { feedSettings } = client.getQueryData<AllTagCategoriesData>(key);
+      updateLocalFeedSettings(feedSettings);
+    }
```

Comment:
What's this?

Reply:
This is setting the local feed filters, this will be soon deprecated since our new system now supports without redirection for authentication.

Reply:
Ok so I'll wait for the new implementation